File: Kingdom-Fall-UI-Enabled/src/main/java/objets/Type_Objet.java
package objets;

public enum Type_Objet {

    ARMES(5, true),
    ARMURES(5, true),
    POTIONS(5, false),
    DIVERS(10, false);

    private final int espaceInventaire;
    private final boolean equipable;

    Type_Objet(int espaceInventaire, boolean equipable) {
        this.espaceInventaire = espaceInventaire;
        this.equipable = equipable;
    }

    public int getEspaceInventaire() {
        return espaceInventaire;
    }

    public boolean isEquipable() {
        return equipable;
    }
}

File: KF-Remake/Kingdom-Fall/source/objets/TypeObjet.cs
namespace objets
{
    public enum TypeObjet
    {
        Armes,
        Armures,
        Potions,
        Divers
    }


    public static class TypeObjetExtensions
    {
        private static readonly Dictionary<TypeObjet, (int espaceInventaire, bool equipable)> _typeData = 
            new Dictionary<TypeObjet, (int, bool)>
            {
                { TypeObjet.Armes, (5, true) },
                { TypeObjet.Armures, (5, true) },
                { TypeObjet.Potions, (5, false) },
                { TypeObjet.Divers, (10, false) }
            };

        public static int GetEspaceInventaire(this TypeObjet type)
        {
            return _typeData[type].espaceInventaire;
        }

        public static bool IsEquipable(this TypeObjet type)
        {
            return _typeData[type].equipable;
        }
    }

}



